### PR TITLE
Added parameters to configure amcl laser scan topic

### PR DESCRIPTION
**Pull request by Michael-Equi**
## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | TB3 simulation |

---

## Description of contribution in a few bullet points

Adds the ability to configure the amcl scan topic in the parameters yaml file like the topics of the rest of the ROS2 nav nodes.  

## Description of documentation updates required from your changes
Added in PR to the configuration docs of navigation.ros.org (https://github.com/ros-planning/navigation.ros.org/pull/45)

### CODE REVIEW

File: nav2_amcl/src/amcl_node.cpp [hunk 1]
```diff
@@ -213,6 +213,9 @@ AmclNode::AmclNode()
     "Requires that AMCL is provided an initial pose either via topic or initial_pose* parameter "
     "(with parameter set_initial_pose: true) when reset. Otherwise, by default AMCL will use the"
     "last known pose to initialize");
+
+  add_parameter("topic", rclcpp::ParameterValue("scan"));
```
Comment (SteveMacenski) [inline nav2_amcl/src/amcl_node.cpp:217]: Add description to this

Comment (SteveMacenski) [reply nav2_amcl/src/amcl_node.cpp:217]: also "topic" is a poor name, we need to specify what this topic is for / composed of. How about `scan_topic`?

**Comment by SteveMacenski** [comment]
Also add to the parameters md file in `/docs/parameters`

File: nav2_amcl/src/amcl_node.cpp (hunk 1 above)
Comment (Michael-Equi) [reply nav2_amcl/src/amcl_node.cpp:217]: I tried to stick with the same convention as the costmap layers but I guess the name spacing makes a difference. Ill go ahead and change it.

**Comment by SteveMacenski** [comment]
@Michael-Equi Please make sure to lint PRs. This is another case where it looks like you didn't run ament_uncrustify + ament_cpplint + ament_cppcheck

```
    --- src/amcl_node.cpp

    +++ src/amcl_node.cpp.uncrustify

    @@ -217 +217,2 @@

    -  add_parameter("scan_topic", rclcpp::ParameterValue("scan"),

    +  add_parameter(

    +    "scan_topic", rclcpp::ParameterValue("scan"),

    

    1 files with cod
```

**Comment by SteveMacenski** [comment]
@Michael-Equi thanks for the contribution, you still need to update your PR on navigation.ros.org that needs a quick update to be merged

**Pull request merged**